cloud/services/domains.go

```diff
@@ -62,6 +73,82 @@ func EnsureDNSEntries(ctx context.Context, mscope *scope.MachineScope, operation
 	return nil
 }
 
+// EnsureAkamaiDNSEntries ensures the domainrecord on Akamai EDGE DNS is created, updated, or deleted based on operation passed
+func EnsureAkamaiDNSEntries(ctx context.Context, mscope *scope.MachineScope, operation string, dnsEntries []DNSOptions) error {
+	// SetUp Client for Akamai EDGE DNS
+
+	fqdn := mscope.LinodeCluster.ObjectMeta.Name + "-" + mscope.LinodeCluster.Spec.Network.DNSUniqueIdentifier + "." + mscope.LinodeCluster.Spec.Network.DNSRootDomain
+
+	for _, dnsEntry := range dnsEntries {
+		recordBody, err := mscope.AkamaiDomainsClient.GetRecord(ctx, mscope.LinodeCluster.Spec.Network.DNSRootDomain, fqdn, string(dnsEntry.DNSRecordType))
+		if err != nil {
+			if operation == "create" {
+				if err := createAkamaiDNSEntry(ctx, mscope, fqdn, dnsEntry); err != nil {
+					return err
+				}
+			}
+			continue
+		}
+		if operation == "delete" {
+			switch {
+			case len(recordBody.Target) > 1:
+				recordBody.Target = removeElement(recordBody.Target, strings.Replace(dnsEntry.Target, "::", ":0:0:", 8)) //nolint:mnd // 8 for 8 octest
+				if err := updateAkamaiDNSEntry(ctx, mscope, recordBody); err != nil {
+					return err
+				}
+				continue
+			default:
+				if err := deleteAkamaiDNSEntry(ctx, mscope, recordBody); err != nil {
+					return err
+				}
+			}
+		} else {
+			recordBody.Target = append(recordBody.Target, dnsEntry.Target)
+			if err := updateAkamaiDNSEntry(ctx, mscope, recordBody); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func removeElement(stringList []string, elemToRemove string) []string {
+	for index, element := range stringList {
+		if element == elemToRemove {
+			stringList = append(stringList[:index], stringList[index+1:]...)
+			continue
+		}
+	}
+	return stringList
+}
+
+func createAkamaiDNSEntry(ctx context.Context, mscope *scope.MachineScope, fqdn string, dnsOptions DNSOptions) error {
+	recordBody := &dns.RecordBody{
+		Name:       fqdn,
+		RecordType: string(dnsOptions.DNSRecordType),
+		TTL:        dnsOptions.DNSTTLSec,
+		Target:     []string{dnsOptions.Target},
+	}
+	if err := mscope.AkamaiDomainsClient.CreateRecord(ctx, recordBody, mscope.LinodeCluster.Spec.Network.DNSRootDomain); err != nil {
+		return err
+	}
+	return nil
+}
+
+func updateAkamaiDNSEntry(ctx context.Context, mscope *scope.MachineScope, recordBody *dns.RecordBody) error {
+	if err := mscope.AkamaiDomainsClient.UpdateRecord(ctx, recordBody, mscope.LinodeCluster.Spec.Network.DNSRootDomain); err != nil {
+		return err
+	}
+	return nil
```

Comment:
Shouldn't need a conditional here
```suggestion
	return mscope.AkamaiDomainsClient.UpdateRecord(ctx, recordBody, mscope.LinodeCluster.Spec.Network.DNSRootDomain)
```

Reply:
done